src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -94,47 +95,50 @@ public T GetObject<T>(T input)
             PropertyInfo[] properties;
             Type typeKey = typeof(T);
 
-            if (!_propertyCache.TryGetValue(typeKey, out properties))
+            lock (_lockObject)
```

Comment:
Looking at references to `MetadataReferenceCache`, I would guess this class is not used in a restore.  Is that right @nkolev92 ?

But I don't like the fact that we're slowing down all usage of this method, just because of one VS scenario. Command line, and other scenarios, shouldn't be punished for VS technical debt, in my opinion.

Reply:
Switched to concurrent dictionary

Reply:
The metadata cache is used by V2 feeds, so technically it will be used by restore.